manta-accounting/src/wallet/signer/functions.rs

```diff
@@ -150,7 +149,7 @@ fn insert_next_item<C>(
     assets: &mut C::AssetMap,
     parameters: &Parameters<C>,
     identified_asset: IdentifiedAsset<C>,
-    nullifiers: &mut Vec<Nullifier<C>>,
+    nullifiers: &mut C::NullifierMap,
```

Comment:
one thing im noticing is that the map doesn't get cleared when doign a resync.
not sure if that's a bad thing though

Reply:
nvm they do get reset

Reply:
actually maybe it's a good idea to release all the nullifiers that are not my own
it might get laggy to carry around 100s of thousand of nullifiers in memory

Reply:
It is time vs space, as usual. If we release them from memory, then next time we sync we have to: pull them all again and insert them all (which is O(n) where n is the total number of nullifiers vs O(m) where m is the number of new nullifiers).

A more interesting question is: can we just keep the Nullifiers that belong to us? I think we can by using the OutgoingNotes but I didn't want to overcomplicate this PR. That's a further optimization for when we are sure the bug is fixed.